.ci/conda-envs/ci-core-py38.txt

```diff
@@ -41,6 +41,7 @@ bokeh=3.1.*
 fsspec=2024.5.*
 msgpack-python=1.0.*
 pluggy=1.5.*
+pyparsing=3.1.4
```

Comment:
While working on this, I discovered https://github.com/microsoft/LightGBM/issues/6680.

Have reported that issue upstream: https://github.com/conda-forge/pyparsing-feedstock/issues/48

For LightGBM's purposes, testing on an old version of Python, let's just pin it as we do other libraries in these environments.